### PR TITLE
Refactors facts about Java version

### PR DESCRIPTION
**Pull request by petems**
These facts have since been refactored, ported these changes from https://github.com/puppetlabs/puppetlabs-java/

### CODE REVIEW

File: lib/facter/java_major_version.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#   none
+#
+# Notes:
+#   None
+Facter.add(:java_major_version) do
+  setcode do
+    java_version = Facter.value(:java_version)
+    java_patch_level = java_version.strip.split('_')[0].split('.')[1] unless java_version.nil?
+  end
+end
```
Comment (mikemcquaid) [inline lib/facter/java_major_version.rb]: Needs a newline.

**Comment by mikemcquaid** [comment]
Hmm, I'm wondering if/why we need this for Java 8 support when we can just do version comparisons on the existing fact?

File: lib/facter/java_major_version.rb (hunk 1 above)
Comment (petems) [inline lib/facter/java_major_version.rb]: Fixed

**Comment by petems** [comment]
@mikemcquaid We don't _need_ it, as you can perform regex's on the current fact, it just makes it a little easier to have Facter/Ruby code to perform that string manipulation :smile:

**Comment by mikemcquaid** [comment]
@petems Could I see some sample code that you'd like to use this? Thanks!

**Comment by petems** [comment]
Been a while, gonna close this

**Pull request closed without merging**